src/core/a-entity.js

```diff
@@ -461,22 +467,23 @@ var proto = Object.create(ANode.prototype, {
       var component = this.components[attr] || components[attr];
       var partialComponentData;
       value = value === undefined ? '' : value;
-      var valueStr = value;
+      var deserializedValue = value;
       var oldValue;
+      var serializedValue = value;
 
       if (component) {
         if (typeof value === 'string' && componentPropValue !== undefined) {
           // Update currently-defined component data with the new property value.
           partialComponentData = self.getAttribute(attr) || {};
           partialComponentData[value] = componentPropValue;
-          value = partialComponentData;
+          deserializedValue = partialComponentData;
```

Comment:
Just reading the code I feel friction trying to remember: Is serialized a string and deseralized an object or the other way around? Can we call them valueObj / valueStr ?